contracts/payloads/IGP7/PayloadIGP7.sol

```diff
@@ -0,0 +1,337 @@
+    ILite public constant LITE = ILite(0xA0D3707c569ff8C87FA923d3823eC5D81c98Be78);
+    IDSAV2 public constant TREASURY = IDSAV2(0x28849D2b63fA8D361e5fc15cB8aBB13019884d09);
+
+    uint256 public constant ONE_DAY_TIME_IN_SECONDS = 1 days; // 1 day in seconds. 86400s
+    uint256 public constant ONE_DAY_TIME_IN_BLOCKS = 7_200; // 1 day in blocks. 12s per block
+    uint256 public constant TWO_DAY_TIME_IN_BLOCKS = 14_400; // 2 day in blocks. 12s per block
+
+    address public immutable GOVERNOR_IMPLEMENTATION_ADDRESS;
+    address public constant TEAM_MULTISIG = 0x4F6F977aCDD1177DCD81aB83074855EcB9C2D49e;
+
+    constructor (address governor_, address timelock_) {
+        TIMELOCK = ITimelock(address(timelock_)); 
+        GOVERNOR_IMPLEMENTATION_ADDRESS = address(governor_);
+        ADDRESS_THIS = address(this);
+    }
+
+
+    function propose(string memory description) external {
+        require(msg.sender == PROPOSER, "msg.sender-not-proposer");
+
+        uint256 totalActions = 8;
+        address[] memory targets = new address[](totalActions);
+        uint256[] memory values = new uint256[](totalActions);
+        string[] memory signatures = new string[](totalActions);
+        bytes[] memory calldatas = new bytes[](totalActions);
+
+        // Action 1: call cast() - transfer rewards to Team Multisig, add new Timelock as auth & remove old Timelock as auth on Treasury
+        (targets[0], values[0], signatures[0], calldatas[0]) = action1();
+
+        // Action 2: call _setImplementation() - upgrade governor contract to new implementation
+        (targets[1], values[1], signatures[1], calldatas[1]) = action2();
+
+        // Action 3: call changeMaster() - change ownership of DSA to new timelock contract
+        (targets[2], values[2], signatures[2], calldatas[2]) = action3();
+
+        // Action 4: call setAdmin() - change ownership of Lite to new timelock contract
+        (targets[3], values[3], signatures[3], calldatas[3]) = action4();
+
+        // Action 5: call _setPendingAdmin() - on governor contract with new timelock address
+        (targets[4], values[4], signatures[4], calldatas[4]) = action5();
+
+        // Action 6: call setPendingAdmin() - on old timelock to change team multisig
+        (targets[5], values[5], signatures[5], calldatas[5]) = action6();
+
+        // Action 7: call queueTransaction - new timelock contract to queue below payload
+        (targets[6], values[6], signatures[6], calldatas[6]) = action7();
+
+        // Action 8: call executeTransaction - new timelock contract to execute below payload
+        (targets[7], values[7], signatures[7], calldatas[7]) = action8();
+
+        uint256 proposedId = GOVERNOR.propose(
+            targets,
+            values,
+            signatures,
+            calldatas,
+            description
+        );
+
+        require(proposedId == PROPOSAL_ID, "PROPOSAL_IS_NOT_SAME");
+    }
+
+    function execute() external {
```

Comment:
proposal on forum talks about 4 days for execution time? 
![image](https://github.com/Instadapp/inst-governance/assets/6112929/038b8c33-1347-4a15-81de-20c3858be63c)

https://gov.instadapp.io/t/update-timelock-contracts-and-transfer-fluid-rewards/684
here it is correct.

Reply:
let me look into it